mobile/src/screens/marketplace.js

```diff
@@ -550,6 +551,10 @@ class MarketplaceScreen extends Component {
     updateExchangeRate(this.state.fiatCurrency[1], 'DAI')
   }
 
+  onWebViewNavigationStateChange = state => {
+    this.setState({ currentDomain: new URL(state.url).hostname })
```

Comment:
It is good to wrap this in `try { ... } catch { ... }` block, unless we are sure that `state.url` will always be a valid URL. `URL` will throw if `state.url` is a malformed URL

Reply:
This state is reported from the browser inside webView, so it is probably always going to be a valid URL...

Adding try catch to be extra sure. Defensive programming FTW!

Reply:
Thanks @sparrowDom. Everything else looks good to me. But Tom may want to take a pass.

Reply:
Yup I am sure he will thanks @shahthepro